Sync Properties Update Revision Block and Create PDF/Modules/Sample.SyncFilePropsAndUpdRevBlockModule.psm1

```diff
@@ -0,0 +1,846 @@
+						associations,
+						/*bom*/null, /*copyBom*/true, // preserve any BOM
+						file.Name, file.FileClass, file.Hidden, // preserve these attributes
+						new vltobj.ByteArray() { Bytes = uploadTicket }
+						);
+				}
+				finally
+				{
+					// if we got here and file is still checked-out, 
+					// something went wrong so undo the checkout.
+					if (file.CheckedOut)
+						file = svcmgr.DocumentService.UndoCheckoutFile(file.MasterId, out downloadTicket);
+				}
+
+				return file;
+			}
+
+			private static object ConvertInvariantStringToObject(string value, vltobj.DataType typ)
+			{
+				if (value == null) return null; // quick exit on null
+
+				// dates need to parsed with invariant culture and converted to local time
+				if (typ == vltobj.DataType.DateTime)
+					return DateTime.Parse(value, System.Globalization.CultureInfo.InvariantCulture).ToLocalTime();
+				// numerics need to be parsed with invariant culture
+				else if (typ == vltobj.DataType.Numeric)
+					return Double.Parse(value, System.Globalization.CultureInfo.InvariantCulture);
+				// bools are straight forward
+				else if (typ == vltobj.DataType.Bool)
+					return Boolean.Parse(value);
+				else // don't do anything for string or image types (image types should not be seen here)
+					return value;
+			}
+		}
+	}
+'@
+#endregion
+
+function SyncProperties
+{
+    param($file)
+
+	Write-Host "Start 'Sync Properties' of file '$($file.Name)' ..."
+	$s = "/AutodeskDM/Services/v"
+	$start = $vault.AdminService.Url.IndexOf($s) + $s.Length
+	$version = [int]$vault.AdminService.Url.Substring($start, 2)
+
+	# 27 = 2022
+	# 25 = 2020
+	# 24 = 2019
+	# 23 = 2018
+	# 22 = 2017
+	# 21 = 2016
+	
+	if ($version -ge 27) {
+		$cs = $2022AndNewerCs
+	} elseif ($version -ge 25) {
+		$cs = $2020AndNewerCs
+	} elseif ($version -ge 22) {
+		$cs = $2017AndNewerCs
+	} else {
+		$cs = $OlderThan2017Cs
+	}
+
+	$null = [System.Reflection.Assembly]::LoadWithPartialName("Autodesk.Connectivity.WebServices")
+	Add-Type -TypeDefinition $cs -ReferencedAssemblies @("Autodesk.Connectivity.WebServices", "System.Core")
+	
+	$propertySync = New-Object Autodesk.Vault.SDKTools.PropertySync -ArgumentList @($vault)
+	$writeResults = New-Object Autodesk.Connectivity.WebServices.PropWriteResults
+	$cloakedEntityClasses = @()
+	$vaultFile = $vault.DocumentService.GetFileById($file.Id)
+	$vaultFile = $propertySync.SyncProperties($vault, $vaultFile, "Property Sync", $true, [ref]$writeResults, [ref]$cloakedEntityClasses, $true)
+	Write-Host "End 'Sync Properties' of file '$($file.Name)'"
+}
+function UpdateRevisionBlock {
+	param ($file)
+
+	#using Vault API to get the file instead of using powerVault's $file since it runs into property sync job not finding any equivalence problems when '$file.Name' or '$file._Name' is used
+	$fileForUpdRevJob = $vault.DocumentService.GetLatestFileByMasterId($file.MasterId)
+
+	Write-Host "Start 'Update Revision Block' of file '$($fileForUpdRevJob.Name)' ..."
```

Comment:
this is probably not needed. I heard that the issue with powerVault's file is not existing anymore. maybe it was fixed with some other fix of powerVault issues.